CheckpointPlugin.cpp

```diff
@@ -595,14 +644,50 @@ bool CheckpointPlugin::rewind(ServerWrapper sw) {
 		return false;  // Ignored whatever inputs may have happened to exit mode; do not apply state.
 	}
 	lastRewindTime = currentTime;
-	int buttonsDown = (abs(ci.Throttle) > 0.1 ? 0x01 : 0) |
-		(abs(ci.Roll) > 0.1 ? 0x02 : 0) |
-		(ci.Handbrake ? 0x04 : 0) |
-		(ci.Jump ? 0x08 : 0) |
-		(ci.ActivateBoost ? 0x10 : 0) |
-		(ci.HoldingBoost ? 0x20 : 0) |
-		((rewindState.atCheckpoint || rewindState.justLoadedQuickCheckpoint) && abs(ci.Steer) >= .05 ? 0x40 : 0) |
-		((rewindState.atCheckpoint || rewindState.justLoadedQuickCheckpoint || abs(ci.Pitch) >= .7) && abs(ci.Pitch) >= .05 ? 0x80 : 0);
+
+	// Get user settings
+	std::string rewindInput = cvarManager->getCvar("rewind_input").getStringValue();
+	std::string matchingAxis = cvarManager->getCvar("matching_axis").getStringValue();
+
+	// Determine the active and matching inputs based on user selection
+	float activeInput = getInputValue(rewindInput, ci);
+	float matchingInput = getInputValue(matchingAxis, ci);
+
+	// Retrieve rewind unpause settings from CVars
+	int buttonsDown = 0;
+	bool isAtCheckpoint = rewindState.atCheckpoint || rewindState.justLoadedQuickCheckpoint;
+
+	auto checkAndSet = [&](const std::string& axis, float axisValue, const char* unpauseSetting, const char* thresholdSetting, int buttonBit) {
+		bool isRewindAxis = (axis == rewindInput);
+		bool isMatchingAxis = (axis == matchingAxis);
+		if (cvarManager->getCvar(unpauseSetting).getBoolValue()) {
+			float effectiveThreshold = cvarManager->getCvar(thresholdSetting).getFloatValue();
+			if (isMatchingAxis && isAtCheckpoint) {
+				effectiveThreshold = 0.8; 
+			}
+			if (!isRewindAxis && fabs(axisValue) > effectiveThreshold) {
+				buttonsDown |= buttonBit;
+			}
+		}
+	};
+
+	// Check analog inputs
+	checkAndSet("throttle", ci.Throttle, "enable_throttle_unpause", "throttle_threshold", 0x01);
+	checkAndSet("roll", ci.Roll, "enable_roll_unpause", "roll_threshold", 0x02);
+	checkAndSet("steer", ci.Steer, "enable_steer_unpause", "steer_threshold", 0x40);
+	checkAndSet("pitch", ci.Pitch, "enable_pitch_unpause", "pitch_threshold", 0x80);
+	checkAndSet("yaw", ci.Yaw, "enable_yaw_unpause", "yaw_threshold", 0x100);
```

Comment:
I'm still not sure I see the value in toggling whether various things that move your car around will unpause.  I think _all_ these should _always_ unpause, _unless_ they are used for rewind (and it's possible to rewind -- i.e. not at a checkpoint).  That would eliminate a ton of variables, configuration, and knobs in the UI.